### PR TITLE
Make a11y test more dynamic

### LINKED ISSUE #2221: Make A11y tests more dynamic
**Is your feature request related to a problem? Please describe.**
Following https://github.com/bbc/simorgh/issues/2155 we run the a11y tests on a hardcoded array of urls, we should make this dynamic to run automatically on all available services and page types

**Describe the solution you'd like**
Make the a11y tests dynamic to run automatically on all available services and page types

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**

### PR DESCRIPTION
**Pull request by EinsteinNjoroge**
Resolves #2221 

**Overall change:** _Run `a11y` test dynamic on all available services and articles from all services_

**Code changes:**

- _Get all service names from Cypress service config._

- _Run `a11y` test dynamic on all available services and articles from all services._

- _Remove hardcoded URLs._

---

- [X] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

**Review by dr3 — COMMENTED** [review]
Rather than using fs here, and reading (almost) all article fixtures in the data folder, could we use `cypress/support/config/services.js`? If you make it CJS and fake `Cypress.env('APP_ENV')` you can read all the services and paths easily. 

Worth checking with @benjaminhobbs what he thinks though

**Comment by jamesbhobbs** [comment]
> Rather than using fs here, and reading (almost) all article fixtures in the data folder, could we use `cypress/support/config/services.js`? If you make it CJS and fake `Cypress.env('APP_ENV')` you can read all the services and paths easily.
> 
> Worth checking with @benjaminhobbs what he thinks though

Definitely do this, otherwise our various testing approaches won't ever be kept in lockstep.

**Review by jamesbhobbs — CHANGES_REQUESTED** [review]
Use the cypress config to get the pages pls

**Comment by EinsteinNjoroge** [comment]
> Rather than using fs here, and reading (almost) all article fixtures in the data folder, could we use `cypress/support/config/services.js`? If you make it CJS and fake `Cypress.env('APP_ENV')` you can read all the services and paths easily.
> 
This is much better than using `fs`, its gonna take more time but I like the approach.

**Comment by EinsteinNjoroge** [comment]
> Rather than using fs here, and reading (almost) all article fixtures in the data folder, could we use `cypress/support/config/services.js`? If you make it CJS and fake `Cypress.env('APP_ENV')` you can read all the services and paths easily.
> 
@benjaminhobbs @dr3 how does this look?

**Review by dr3 — COMMENTED** [review]
This is gunna run ~400 times (40 services * 2 pages * 5 breakpoints), seems kinda crazy, could we maybe randomly pick a subset to run on?

File: a11y.js
Comment (jamesbhobbs) [inline a11y.js]: this is unnecessary, can we make this an option to run on all widths and by default only run on 1 mobile width (the one that's most popular)

**Review by jamesbhobbs — DISMISSED** [review]

**Comment by ghost** [comment]
@dr3  
> could we maybe randomly pick a subset to run on?

This will be non-deterministic. i.e Currently Thai frontpage is failing the `ally` test. If `Thai` is not included in the random subset we will have passing tests. If Included it would fail.

Screenshot 2019-08-15 at 15 08 30

**Comment by EinsteinNjoroge** [comment]
@sadickisaac 🤔 somehow `bbca11y` fails on localhost but they pass on Travis and Jenkins

Screen Shot 2019-08-15 at 3 44 02 PM

**Comment by ghost** [comment]
@EinsteinCarrey interesting. However my main concern is having a passing `a11y` test while there is a possibility of one or more services failing. We would not be able to notice this early enough if our sample list doesn't contain the failing service(s).

**Comment by dr3** [comment]
Can we run just the smoke urls? we recently added a smoke key to the cypress data https://github.com/bbc/simorgh/blob/latest/cypress/support/config/services.js#L46

**Comment by EinsteinNjoroge** [comment]
> @EinsteinCarrey interesting. However my main concern is having a passing `a11y` test while there is a possibility of one or more services failing. We would not be able to notice this early enough if our sample list doesn't contain the failing service(s).

@sadickisaac  Makes sense, Drew's suggestion should solve for this

**Review by dr3 — DISMISSED** [review]
This LGTM, but the a11y test fails, so we cant merge this yet :( Can you write issues to fix the a11y issues, and block this on them please?

**Comment by EinsteinNjoroge** [comment]
blocked by https://github.com/bbc/simorgh/issues/3200

**Comment by EinsteinNjoroge** [comment]
> This LGTM, but the a11y test fails, so we cant merge this yet :( Can you write issues to fix the a11y issues, and block this on them please?

@dr3 I've opened https://github.com/bbc/simorgh/issues/3200 to fix the failing test issue on local but I'm not clear on why it should block this PR, I think its a different issue altogether.

**Comment by dr3** [comment]
@EinsteinCarrey `npm run bbcA11y` fails for me locally on this branch, any idea why its passing on jenkins?

**Comment by EinsteinNjoroge** [comment]
@dr3 Following https://github.com/bbc/bbc-a11y/issues/296, I think we can safely unblock this PR since fixing that issue is outside the scope of this PR.

File: a11y.js
Comment (ghost) [inline a11y.js]:
```suggestion
// This functionality can be extended to allow for testing on all widths
```

**Review by ghost — APPROVED** [review]

File: a11y.js
Comment (j-pendlebury) [inline a11y.js]:
```suggestion
// Run a11y on 360px only since designs are done in this width
```

**Review by j-pendlebury — APPROVED** [review]
just one suggestion. Otherwise looks good to me 👍

**Pull request merged**
